src/QsCompiler/Core/Transformations/NamespaceTransformation.fs

```diff
@@ -84,41 +81,49 @@ type NamespaceTransformationBase internal (options: TransformationOptions, _inte
             let t = this.Statements.Expressions.Types.OnType itemType
             QsTupleItem << Anonymous |> Node.BuildOr original t
         | QsTupleItem (Named item) as original ->
-            let loc = item.Position, item.Range
-            let name = this.OnItemName item.VariableName
+            let loc = this.Common.OnSymbolLocation(item.Position, item.Range)
+            let name = this.Common.OnItemNameDeclaration item.VariableName
             let t = this.Statements.Expressions.Types.OnType item.Type
             let info = this.Statements.Expressions.OnExpressionInformation item.InferredInformation
 
-            QsTupleItem << Named << LocalVariableDeclaration<_>.New info.IsMutable
+            QsTupleItem << Named << LocalVariableDeclaration.New info.IsMutable
             |> Node.BuildOr original (loc, name, t, info.HasLocalQuantumDependency)
 
+    // TODO: RELEASE 2022-09: Remove.
+    [<Obsolete "Use SyntaxTreeTransformation.OnLocalNameDeclaration or override OnArgumentTuple instead.">]
     abstract OnArgumentName : QsLocalSymbol -> QsLocalSymbol
 
-    default this.OnArgumentName arg =
+    // TODO: RELEASE 2022-09: Remove.
+    [<Obsolete "Use SyntaxTreeTransformation.OnLocalNameDeclaration or override OnArgumentTuple instead.">]
+    override this.OnArgumentName arg =
         match arg with
-        | ValidName name -> ValidName |> Node.BuildOr arg (this.Statements.OnVariableName name)
+        | ValidName name -> this.Statements.Expressions.Common.OnLocalNameDeclaration name |> ValidName
         | InvalidName -> arg
 
+    // TODO: RELEASE 2022-09: Remove member.
+    [<Obsolete "Use SyntaxTreeTransformation.OnArgumentTuple instead.">]
     abstract OnArgumentTuple : QsArgumentTuple -> QsArgumentTuple
 
-    default this.OnArgumentTuple arg =
+    // TODO: RELEASE 2022-09: Make this an internal member. Keep the following comment:
+    // do not expose this - this handle is exposed as a virtual member in the SyntaxTreeTransformation itself
+    [<Obsolete "Use SyntaxTreeTransformation.OnArgumentTuple instead.">]
+    override this.OnArgumentTuple arg =
```

Comment:
How we need to go about removing obsolete functions is documented in the form of comments in the code. 
The way we go about this indeed is required to properly deprecate the old handle;
If I had the implementation in the new handle and the old handle pointed to it, then inheriting classes that override the old handle would break - thus we would not have proper deprecation. The pattern that we instead follow is that the implementation remains in the old handle until it is removed, the new handle points to the old handle, and all usages in our transformation infrastructure use the new handle. That way both overriding the old hand and overriding the new handle works properly.

Reply:
That seems inconsistent with this functions references to `this.Common.OnArgumentTuple`. Should it actually reference itself recursively, as it did before? With the new change, I don't see the recursion happening.

Reply:
I am not sure I follow the statement about the inconsistency?
Following my explanation above, here is how this looks like concretely for OnArgumentTuple:

old handle: Namespaces.OnArgumentTuple
new handle: SyntaxTree.OnArgumentTuple ("paraphrasing" a bit here, to keep it concise, let me know if I need to be more precise)

The implementation is in the old handle, and (like for other usages of the handle in the transformation framework) it calls into the new handle. The new handle points back to the old handle. So the recursion is Namespaces.OnArgumentTuple -> SyntaxTree.OnArgumentTuple -> Namespaces.OnArgumentTuple -> ...

We inject a node here in order to properly deprecate the old handle; we replace one node Namespaces.OnArgumentTuple with two nodes SyntaxTree.OnArgumentTuple -> Namespaces.OnArgumentTuple, and that goes for all its uses. If the recursion was direct, then the old handle wouldn't be deprecated properly. Does that clarify things?

Reply:
I see, that makes sense. Thanks for the explanation!